common/nexus/endpoint_registry.go

```diff
@@ -53,8 +54,10 @@ type (
 	}
 
 	EndpointRegistry interface {
-		GetByName(ctx context.Context, endpointName string) (*persistencespb.NexusEndpointEntry, error)
-		GetByID(ctx context.Context, id string) (*persistencespb.NexusEndpointEntry, error)
+		// GetByName returns an endpoint entry for the endpoint name for a caller from the given namespace ID.
+		// Note that the default implementation is global to the cluster and can ignore the namespace ID param.
+		GetByName(ctx context.Context, namespaceID namespace.ID, endpointName string) (*persistencespb.NexusEndpointEntry, error)
```

Comment:
I'm once again annoyed that Go doesn't have optional parameters or method overloading. I think it might be confusing why namespace ID is a required parameter. Maybe we should define a struct for the parameters so not all need to be filled in? This is fine too though.